From 8c8a155f50a4804c658b6976082fc0aac1dd5a09 Mon Sep 17 00:00:00 2001
From: tenzap <fabstz-it@yahoo.fr>
Date: Tue, 2 May 2023 10:07:43 +0200
Subject: [PATCH 2/2] Fix build failure with Qt5.6

With Qt 5.6, build fails with error below.

This is because in Qt 5.6, the 3rd argument is not optional. Starting from Qt
5.7 the default value for the 3rd argument is nullptr, so setting it to
nullptr.

https://doc.qt.io/archives/qt-5.6/qaction.html#QAction-2
https://doc.qt.io/archives/qt-5.7/qaction.html#QAction-2

Error:
src/autotype/AutoTypeSelectDialog.cpp:330:34: error: no matching constructor for initialization of 'QAction'
    auto typeVirtualAction = new QAction(icons()->icon("auto-type"), tr("Use Virtual Keyboard"));
                                 ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
---
 src/autotype/AutoTypeSelectDialog.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/autotype/AutoTypeSelectDialog.cpp b/src/autotype/AutoTypeSelectDialog.cpp
index 71542438..1e767d79 100644
--- src/autotype/AutoTypeSelectDialog.cpp
+++ src/autotype/AutoTypeSelectDialog.cpp
@@ -327,7 +327,7 @@ void AutoTypeSelectDialog::buildActionMenu()
     });
 
 #if defined(Q_OS_WIN) || defined(Q_OS_MAC)
-    auto typeVirtualAction = new QAction(icons()->icon("auto-type"), tr("Use Virtual Keyboard"));
+    auto typeVirtualAction = new QAction(icons()->icon("auto-type"), tr("Use Virtual Keyboard"), nullptr);
     m_actionMenu->insertAction(copyUsernameAction, typeVirtualAction);
     typeVirtualAction->setShortcut(Qt::CTRL + Qt::Key_4);
     connect(typeVirtualAction, &QAction::triggered, this, [&] {
-- 
2.30.2

